src/realm/object-store/results.cpp

```diff
@@ -1080,6 +1080,12 @@ Results Results::import_copy_into_realm(std::shared_ptr<Realm> const& realm)
     util::CheckedUniqueLock lock(m_mutex);
     if (m_mode == Mode::Empty)
         return *this;
+
+    if (!is_valid()) {
```

Comment:
OK, so we know that the result is actually invalid, and we fake an empty result. Probably we need to put a comment somewhere because this is really counter-intuitive.

Reply:
Well, it was simply enough to fix one mentioned usecase, but it seems that the problem is much bigger, since there is no defined behaviour for other types of results after owning entities removal. I'm not sure if it's enough for results with other modes. Should be tested fully.

Reply:
This does not compute. You realize that in case of the crash `is_valid` still returns true here. So this code-path will not be used.. or am I missing something?

Reply:
It is 'true' before obj delete, 'false' after, and then with a fix and empty Results, the new frozen Results returns also 'true', because they are in a sense... (at least if we follow original intention in a comment for this method)